### PR TITLE
Date range support for bible ranges.

### PR DESCRIPTION
**Pull request by ruschein**

### CODE REVIEW

File: java/solr_plugins/RangedSearches/src/de/uni_tuebingen/ub/ixTheo/bibleDateRangeSearch/BibleDateRange.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
+package de.uni_tuebingen.ub.ixTheo.bibleDateRangeSearch;
+
+
+import java.util.ArrayList;
+import java.util.List;
+import de.uni_tuebingen.ub.ixTheo.rangeSearch.Range;
+
+
+public class BibleDateRange extends Range {
+    private final static int VERSE_LENGTH = 3;
```
Comment (mtrojan-ub) [inline java/solr_plugins/RangedSearches/src/de/uni_tuebingen/ub/ixTheo/bibleDateRangeSearch/BibleDateRange.java:10]: same constants as in bibleRangeSearch\bibleRange.java, could they be re-used?

File: java/solr_plugins/RangedSearches/src/de/uni_tuebingen/ub/ixTheo/bibleDateRangeSearch/BibleDateRange.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    private final static int LOWER_BIBLE_CODE_START = 0;
+    private final static int UPPER_BIBLE_CODE_START = LOWER_BIBLE_CODE_START + BIBLE_CODE_LENGTH + 1;
+
+    private final static int RANGE_STRING_LENGTH = BIBLE_CODE_LENGTH + 1 + BIBLE_CODE_LENGTH;
+
+    public BibleDateRange(final String range) {
+        super(getLower(range), getUpper(range));
+    }
+
+    private static long getLower(final String range) {
```
Comment (mtrojan-ub) [inline java/solr_plugins/RangedSearches/src/de/uni_tuebingen/ub/ixTheo/bibleDateRangeSearch/BibleDateRange.java:31]: ...same for these functions

File: java/solr_plugins/RangedSearches/src/de/uni_tuebingen/ub/ixTheo/bibleDateRangeSearch/BibleDateRangeParserPlugin.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import org.apache.solr.request.SolrQueryRequest;
+import org.apache.solr.search.QParser;
+import org.apache.solr.search.QParserPlugin;
+
+
+public class BibleDateRangeParserPlugin extends QParserPlugin {
+
+    @Override
+    @SuppressWarnings("rawtypes")
+    public void init(final NamedList args) {
```
Comment (mtrojan-ub) [inline java/solr_plugins/RangedSearches/src/de/uni_tuebingen/ub/ixTheo/bibleDateRangeSearch/BibleDateRangeParserPlugin.java:15]: at least call super here?

File: java/solr_plugins/RangedSearches/src/de/uni_tuebingen/ub/ixTheo/bibleDateRangeSearch/BibleDateRangeWeight.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+import org.apache.lucene.util.FixedBitSet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.io.IOException;
+import de.uni_tuebingen.ub.ixTheo.rangeSearch.Range;
+import de.uni_tuebingen.ub.ixTheo.rangeSearch.RangeScorer;
+
+
+public class BibleDateRangeWeight extends ConstantScoreWeight {
+    private final static float NOT_RELEVANT = Float.NEGATIVE_INFINITY;
```
Comment (mtrojan-ub) [inline java/solr_plugins/RangedSearches/src/de/uni_tuebingen/ub/ixTheo/bibleDateRangeSearch/BibleDateRangeWeight.java:21]: Also same as BibleRangeWeight.java

File: java/solr_plugins/RangedSearches/src/de/uni_tuebingen/ub/ixTheo/bibleDateRangeSearch/BibleDateRangeParserPlugin.java (hunk 3 above)
Comment (ruschein) [reply java/solr_plugins/RangedSearches/src/de/uni_tuebingen/ub/ixTheo/bibleDateRangeSearch/BibleDateRangeParserPlugin.java:15]: We didn't need this in BibleRangeParserPlugin.java, so I'm not sure if this is needed/helpful here...  I mean then we wouldn't have needed to override the parent class' init in the first place.

File: java/solr_plugins/RangedSearches/src/de/uni_tuebingen/ub/ixTheo/timeAspectDateRangeSearch/TimeAspectDateRangeWeight.java [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+            };
+            return new RangeScorer(this, scoring, twoPhase);
+        } else {
+            return null;
+        }
+    }
+
+    @Override
+    public final boolean isCacheable(LeafReaderContext context) {
+        return false;
```
Comment (mtrojan-ub) [inline java/solr_plugins/RangedSearches/src/de/uni_tuebingen/ub/ixTheo/timeAspectDateRangeSearch/TimeAspectDateRangeWeight.java:97]: could this also leverage performance?

**Pull request merged**